src/filholes.jl

```diff
@@ -0,0 +1,58 @@
+"""
+    fillhole(img; [dims])
+    fillhole(img; se)
+
+Fill the holes in image 'img'. Could be binary or grascale
+
+The `dims` keyword is used to specify the dimension to process by constructing the box shape
+structuring element [`strel_box(img; dims)`](@ref strel_box). For generic structuring
+element, the half-size is expected to be either `0` or `1` along each dimension.
+
+The output has the same type as input image
+"""
+
+function fillhole(img; dims=coords_spatial(img))
+    return fillhole(img, strel_box(img, dims))
+end
+
+function fillhole(img, se)
+    return fillhole!(similar(img), img, se)
+end
+
+function fillhole!(out, img; dims=coords_spatial(img))
+    return fillhole!(out, img, strel_box(img, dims))
+end
+
+function fillhole!(out, img, se)
+    return _fillhole!(out, img, se)
+end
+
+function _fillhole!(out, img, se)
```

Comment:
We should consider typing the arguments here and extracting the number of dimensions and element type from the `AbstractArray{T,N}`.

Reply:
I agree i could but why?
The function is generic, the speedup of binary cases could be to specialize mreconstruct for binary cases
Do you want to specialize this function at fillhole level ?